cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManager.java

```diff
@@ -181,33 +184,34 @@ public String getFileHash(Path relativePath, String commitHash) throws IOExcepti
       // Find the node representing the exact file path in the tree.
       TreeWalk walk = TreeWalk.forPath(git.getRepository(), realRelativePath.toString(), commit.getTree());
       if (walk == null) {
-        throw new NotFoundException("File not found in Git revision tree.");
+        throw new NotFoundException(
+          "File not found in Git revision tree. Try re-cloning remote if file was added recently.");
```

Comment:
This exception may pop up to user and user has no way to do this suggestion. Let's not add a suggestion user can't do.

Reply:
Removed.